osf/management/commands/check_crossref_dois.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import requests
+
+from datetime import timedelta
+
+from framework.celery_tasks import app as celery_app
+from website import settings
+from website import mails
+from django.utils import timezone
+from django.core.management.base import BaseCommand
+
+import django
+django.setup()
+
+from osf.models import Preprint
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def check_crossref_dois(dry_run=True):
+    """
+    This script is to check for any DOI confirmation messages we may have missed during downtime and alert admins to any
+    DOIs that have been pending for X number of days.
+    :param dry_run:
+    :return:
+    """
+    # Create one enormous url to check if all our pending crossref DOIs are good, then set all identifers
+    preprints_with_pending_dois = Preprint.objects.filter(preprint_doi_created__isnull=True,
+                                                      is_published=True)
+
+    if not preprints_with_pending_dois:
+        return
+
+    pending_dois = []
+
+    for preprint in preprints_with_pending_dois:
+        prefix = preprint.provider.doi_prefix
+        pending_dois.append('doi:{}'.format(settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)))
+
+    url = '{}works?filter={}'.format(settings.CROSSREF_JSON_API_URL, ','.join(pending_dois))
+
+    try:
+        resp = requests.get(url)
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        logger.error('Could not contact crossref to check for DOIs, response returned with exception {}'.format(exc))
+        raise exc
+
+    preprints = resp.json()['message']['items']
+
+    for preprint in preprints:
+        guid = preprint['DOI'].split('/')[-1]
+        pending_preprint = preprints_with_pending_dois.get(guids___id=guid)
+        if not dry_run:
+            pending_preprint.set_identifier_values(preprint['DOI'])
```

Comment:
If I understand to correctly, there's no real need to keep DOI specific info like that, Crossref keeps that info in fine grained detail in it's API and it will reliably be there if we need it to debug.

Second, 👍